test/compress/drop-unused.js

```diff
@@ -3081,13 +3081,37 @@ issue_4235: {
     }
     expect: {
         void function() {
-            var f;
-            console.log(f);
+            var f = console.log(f);
```

Comment:
Interesting. The assignment is harmless in this particular test case because `f` remains undefined before the call and `f` is not used after the call. But in general since uglify-js doesn't know what `console.log()` returns it can't usually make this assumption. What's the motivation in this particular case to join the var declaration and statement?

Reply:
In this case `compress` didn't join them; it just didn't split the original input under `unused` (yet).

Reply:
Sorry, I don't quite follow. Which compress option introduced the assignment in this case and what was the motivation?

Reply:
... and if you mean why the original case has that assignment (which seems to be the more sensible question 😅), it's because it clobbers `unused` enough to trigger a bug with multi-declared variables.

Reply:
Crap - this is the **expect** clause - not the input.  Sorry for my confusion.

Reply:
The original reduced test case in #4235 may make this more apparent:
```js
// reduced test case (output will differ)

// (beautified)
function f0() {
    {
        const b_2 = 0;
    }
    (function b_2() {
        var b_2 = b_2[0];
    })();
}

f0();
// output: TypeError: Cannot read property '0' of undefined
// minify: ReferenceError: b_2 is not defined
// options: {
//   "mangle": false,
//   "validate": true
// }
```
The `console.log()` is merely inserted for the benefit of `expect_stdout` cleanliness.

Reply:
Yes, indeed. My Diff-Fu is weak until I've had a coffee.